onert-micro/onert-micro/include/core/OMRuntimeData.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ONERT_MICRO_CORE_RUNTIME_DATA_H
+#define ONERT_MICRO_CORE_RUNTIME_DATA_H
+
+#include "OMTensorData.h"
+#include "OMUtils.h"
+
+#include <memory>
+#include <type_traits>
+
+namespace onert_micro
+{
+namespace core
+{
+
+// clang-format off
+
+// ------------------------------------------------------------------------------------------------
+
+class OMBaseContext
+{
+protected:
+  const circle::Tensor *_tensor;
+  OMRuntimeShape _shape;
```

Comment:
IMHO shape looks redundant.
We can retrieve shape using `OMRuntimeShape(_tensor)`.